DesignDoc.md

```diff
@@ -0,0 +1,157 @@
+
+2. As a user, I want to be able to search for movies by title, director, or actor so that I can easily find the films I'm looking for.
+
+3. As a user, I want to be able to rate and review movies on Reel Deal so that I can share my thoughts with others.
+
+4. As a user, I want to be able to follow other users on Reel Deal so that I can see their reviews and ratings, and discover new movies.
+
+5. As a user, I want to be able to save movies to my "watchlist" on Reel Deal so that I can keep track of the films I want to watch later.
+
+6. As a user, I want to be able to create custom lists on Reel Deal so that I can organize my favorite films by genre, director, or any other criteria.
+
+7. As a user , I want to be able to view statistics and data on movies on Reel Deal so that I can learn more about the films I watch.
+
+8. As a user, I want to be able to view trailers and posters for movies on Reel Deal so that I can get a better idea of what a film is about before watching it.
+
+
+### 4.1. In Scope
+
+* looking at reviews from others people
+* search movies by title director and actor
+* show popular movies to leave reviews on.
+* Adding, updating, deleting, and viewing reviews.
+
+
+### 4.2. Out of Scope
+
+* following other users .
+
+
+# 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
+
+We will use API Gateway and Lambda to create endpoints that will handle the creation, updating, and retrieval of movies and reviews. 
+
+We will store movie review information in a table in DynamoDB.
+
+ReelDeal will also provide a web interface for users. A main page providing a variety of options to add, update and reminisce of all the movies you watched and maybe some you warn others not to watch.
+
+
+# 6. API
+
+## 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package._
+
+```
+// Movie 
+    private int id;
+    private String title;
+    private String description;
+    private String releaseDate;
+    private String posterUrl;
+    private List<String> genres;
+    private List<String> cast;
+    private String director;
+
+
+// Review 
+    private int id;
+    private int movieId;
+    private String username;
+    private String text;
+    private int rating;
+    private Date date;
+
+
+// User 
+    private int id;
+    private String username;
+    private String password;
+    private String email;
+    private List<Review> reviews;
+
+
+
+```
+
+GET /movies - Retrieve a list of all movies
```

Comment:
Please flesh out these endpoints similar to how they were done in the midstone [example design document](https://github.com/nss-se-cohort-1/u5-ProjectTemplate/blob/main/resources/example-design-document.md). E.g. What's the name of the endpoint, the method (which you have here), the url (which you have here), a brief description (which you have here), data types/shapes where appropriate, exceptions thrown.  Aside from that I think this is a pretty thorough list of what you'll need.

Reply:
Thanks for updating these. For the POST/PUT endpoints please add a description of the data shape. E.g. `{"title": "movie title", "description": "description of movie", ...}` for creating a movie.